### PR TITLE
feat(connector): sink support for elasticsearch

### PR DESCRIPTION
**Pull request by wugouzi**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Add elasticsearch sink support for connector.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: java/connector-node/risingwave-sink-es/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            <version>${jackson.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+            <version>${jackson.version}</version>
+        </dependency>
+
+        <!-- ElasticSearch drivers -->
+        <dependency>
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-es/pom.xml:54]: We may move the dependency with version to dependency management of the parent pom, and here we only set the group id and artifact id.

File: java/connector-node/risingwave-sink-es/pom.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            <artifactId>elasticsearch</artifactId>
+            <version>7.17.10</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>elasticsearch-rest-high-level-client</artifactId>
+            <version>7.17.10</version>
+        </dependency>
+        <dependency>
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-es/pom.xml:65]: Where do we depend on this dependency?

**Review by wenym1 — COMMENTED** [review]
Rest LGTM. Thanks for the PR!

File: java/connector-node/risingwave-sink-es/pom.xml (hunk 2 above)
Comment (wugouzi) [reply java/connector-node/risingwave-sink-es/pom.xml:65]: We are using high level client to send the request.

Comment (wenym1) [reply java/connector-node/risingwave-sink-es/pom.xml:65]: I mean `org.apache.httpcomponents:httpcore`

Comment (wugouzi) [reply java/connector-node/risingwave-sink-es/pom.xml:65]: It provides `HttpHost` which is used by es api to build client.

File: java/connector-node/risingwave-connector-test/src/test/java/com/risingwave/connector/sink/elasticsearch/EsSinkTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.junit.Test;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+public class EsSinkTest {
+
+    static TableSchema getTestTableSchema() {
```
Comment (wenym1) [inline java/connector-node/risingwave-connector-test/src/test/java/com/risingwave/connector/sink/elasticsearch/EsSinkTest.java:44]: As I remember, there is a similar method in the a existing util class. We may use that class.

File: java/connector-node/risingwave-sink-es/src/main/java/com/risingwave/connector/EsSink.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+ *
+ * 4. bulkprocessor and high-level-client are deprecated in es 8 java api.
+ */
+public class EsSink extends SinkBase {
+    private static final String ERROR_REPORT_TEMPLATE = "Error when exec %s, message %s";
+
+    private final EsSinkConfig config;
+    private final BulkProcessor bulkProcessor;
+    private final RestHighLevelClient client;
+    private static final Logger LOG = LoggerFactory.getLogger(EsSink.class);
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-es/src/main/java/com/risingwave/connector/EsSink.java:60]: nit: usually the `LOG` or static final field is put at the beginning of the class.

File: java/connector-node/risingwave-sink-es/src/main/java/com/risingwave/connector/EsSink.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+     * @param row
+     * @return
+     */
+    private String buildId(SinkRow row) {
+        String id;
+        if (primaryKeyIndexes.isEmpty()) {
+            id = row.get(0).toString();
+        } else {
+            id = row.get(primaryKeyIndexes.get(0)).toString();
+            for (int i = 1; i < primaryKeyIndexes.size(); i++) {
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-es/src/main/java/com/risingwave/connector/EsSink.java:186]: We may use join to avoiding using a for loop.

**Review by wenym1 — APPROVED** [review]
Rest LGTM

File: java/connector-node/risingwave-connector-test/src/test/java/com/risingwave/connector/sink/elasticsearch/EsSinkTest.java (hunk 3 above)
Comment (wugouzi) [reply java/connector-node/risingwave-connector-test/src/test/java/com/risingwave/connector/sink/elasticsearch/EsSinkTest.java:44]: Here the primary keys are different to test the generated ids.

**Pull request merged**